src/sources/generator.rs

```diff
@@ -314,4 +318,18 @@ mod tests {
         }
         assert_eq!(rx.try_recv(), Err(mpsc::error::TryRecvError::Closed));
     }
+
+    #[tokio::test]
+    async fn json_format_generates_output() {
+        let mut rx = runit(
+            r#"format = "json"
+            count = 5"#,
+        )
+        .await;
+
+        for _ in 0..5 {
+            assert!(matches!(rx.try_recv(), Ok(_)));
```

Comment:
This would be a fairly simple one to ensure the output format is correct, as you can just pass it through `serde_json::from_str::<Value>(&line).unwrap()`.

Reply:
Good call. Done.